lib/CXGN/Cross.pm

```diff
@@ -46,19 +46,30 @@ sub get_cross_relationships {
     my $maternal_parent = "";
     my $paternal_parent = "";
     my @progeny = ();
+    my @filtered_progeny = ();
 
     foreach my $child ($crs->all()) {
         if ($child->type->name() eq "female_parent") {
-          $maternal_parent = [ $child->subject->name, $child->subject->stock_id() ];
+            $maternal_parent = [ $child->subject->name, $child->subject->stock_id() ];
         }
         if ($child->type->name() eq "male_parent") {
-          $paternal_parent = [ $child->subject->name, $child->subject->stock_id() ];
+            $paternal_parent = [ $child->subject->name, $child->subject->stock_id() ];
         }
         if ($child->type->name() eq "offspring_of") {
-          push @progeny, [ $child->subject->name, $child->subject->stock_id() ];
+            my $progeny = $child->subject->stock_id();
+
+            my $offspring_rs = $self->bcs_schema->resultset("Stock::Stock")->search ({
```

Comment:
instead of another search, you can filter the obsolete stocks directly 

my $offspring_rs = $child->search_related('stocks', is_obsolete=> {'!=' => 't' } );

or just check if the progeny is obsolete

  my $is_progeny_obsolete = $child->subject->is_obsolete();
if ($is_progeny_obsolete eq 't') { 
    ## do nothing;
} else { 
   push @filtered_progeny ..... 
 }